packages/@textlint/kernel/src/fixer/fixer-processor.ts

```diff
@@ -69,9 +69,9 @@ export default class FixerProcessor {
             return (sourceText: string): Promise<string> => {
                 // create new SourceCode object
                 const preProcessResult = preProcess(sourceText, sourceCode.filePath);
-                const isPluginReturnAnAST = isTxtAST(preProcessResult);
-                const textForAST = isPluginReturnAnAST ? sourceText : preProcessResult.text;
-                const ast = isPluginReturnAnAST ? preProcessResult : preProcessResult.ast;
+                const isParsedObject = isPluginParsedObject(preProcessResult);
```

Comment:
mitigate checking of the AST that is returned by plugin.

It aims to improve perf.